File: backend/app/schemas/analytics.py
from pydantic import BaseModel
from typing import Any


class BreakdownItem(BaseModel):
    dimension: str | None
    total_requests: int
    total_prompt_tokens: int | None
    total_completion_tokens: int | None
    total_cost_usd: float | None
    avg_latency_ms: float | None


class SummaryResponse(BaseModel):
    total_requests: int
    total_prompt_tokens: int
    total_completion_tokens: int
    total_cost_usd: float
    avg_latency_ms: float
    p50_latency_ms: float
    p95_latency_ms: float
    error_count: int
    breakdown: list[dict[str, Any]]


class TimeseriesResponse(BaseModel):
    data: list[dict[str, Any]]


class DimensionsResponse(BaseModel):
    providers: list[str]
    models: list[str]
    masked_api_keys: list[str]
    origin_domains: list[str]
    labels: list[str]

File: frontend/src/types/analytics.ts
export interface SummaryResponse {
  total_requests: number;
  total_prompt_tokens: number;
  total_completion_tokens: number;
  total_cost_usd: number;
  avg_latency_ms: number;
  p50_latency_ms: number;
  p95_latency_ms: number;
  error_count: number;
  breakdown: BreakdownItem[];
}

export interface BreakdownItem {
  dimension: string | null;
  total_requests: number;
  total_prompt_tokens: number | null;
  total_completion_tokens: number | null;
  total_cost_usd: number | null;
  avg_latency_ms: number | null;
}

export interface TimeseriesPoint {
  bucket: string;
  dimension?: string;
  total_requests: number;
  total_cost_usd: number | null;
  total_prompt_tokens: number | null;
  total_completion_tokens: number | null;
  avg_latency_ms: number | null;
}

export interface TimeseriesResponse {
  data: TimeseriesPoint[];
}

export interface DimensionsResponse {
  providers: string[];
  models: string[];
  masked_api_keys: string[];
  origin_domains: string[];
  labels: string[];
}

export interface AnalyticsFilters {
  start_date?: string;
  end_date?: string;
  group_by?: string;
  granularity?: "hour" | "day" | "week";
}
